controllers/agents/app/servicebinding_controller.go

```diff
@@ -376,15 +376,15 @@ func (r *ServiceBindingReconciler) getApplication(ctx context.Context,
 	sb primazaiov1alpha1.ServiceBinding) ([]unstructured.Unstructured, error) {
 	var applications []unstructured.Unstructured
 	l := log.FromContext(ctx)
-	if sb.Spec.Application.Name != "" {
-		applicationLookupKey := client.ObjectKey{Name: sb.Spec.Application.Name, Namespace: sb.Namespace}
+	if sb.Spec.Application.Selector.ByName != "" {
```

Comment:
In service binding maintain compatibility with servicebinding spec. Prefer `name` over `labels`. If none is defined, add an error condition